internal/manager/run.go

```diff
@@ -162,6 +162,28 @@ func Run(ctx context.Context, c *Config, diagnostic util.ConfigDumpDiagnostic) e
 		return err
 	}
 
+	if featureGates[gatewayFeature] {
+		setupLog.Info("Gateway Feature Enabled, Checking Existence of CRDs")
+		for _, checker := range gatewayBetaCRDsExistsCheckers {
+			if !checker.CRDExists(mgr.GetClient()) {
+				setupLog.Info("CRD does not exist, disable gateway feature", "resource", checker.GVR.Resource)
+				featureGates[gatewayFeature] = false
+				break
+			}
+		}
+	}
+
+	// Check alpha CRDs if GatewayAlpha enabled.
+	if featureGates[gatewayAlphaFeature] {
+		setupLog.Info("GatewayAlpha Feature Enabled, Checking Existence of CRDs")
+		for _, checker := range gatewayAlphaCRDsExistsCheckers {
+			if !checker.CRDExists(mgr.GetClient()) {
+				setupLog.Info("Warning: GatewayAlpha feature enabled but CRD does not exist",
+					"resource", checker.GVR.Resource)
```

Comment:
Shouldn't we disable the gateway alpha features flag here: `gatewayAlphaFeature`? Or do we assume that when someone enabled a "disabled by default" flag he knows what he's doing and he'll install the CRDs later on?

Reply:
Discussed it with @shaneutt here: https://github.com/Kong/kubernetes-ingress-controller/pull/2815#discussion_r945832445. We could assume that the user knows what is going on. What's more, the alpha resources have no dependencies or references of each other, and our `AutoHandler` has already prevented the errors happened on watching non-exist resources.